linera-execution/src/execution.rs

```diff
@@ -215,9 +215,10 @@ where
             UserAction::Message(context, message) => {
                 contract.execute_message(context, runtime_sender, message)
             }
-        };
+        });
         let (runtime_result, call_result) =
             futures::future::join(runtime_actor.run(), call_result_future).await;
+        let call_result = call_result?;
```

Comment:
Nit: I think the join isn't needed anymore. Now you can do:

```rust
let runtime_result = runtime_actor.run().await;
call_result_future.await?;
runtime_result?;
```

Reply:
good catch!